### PR TITLE
add Catch 2.0.1

### PR DESCRIPTION
**Pull request by dvirtz**

### CODE REVIEW

**Comment by maddouri** [comment]
It seems that `examples/Catch` compiles and runs without problems by removing [`examples/Catch/CMakeLists.txt:11`](https://github.com/dvirtz/hunter/blob/5d8949c5c2ebd8d4890bfc6788109bad69ffeb66/examples/Catch/CMakeLists.txt#L11)

```cpp
set(CMAKE_CXX_STANDARD 98)
```

then adding

```cpp
set_target_properties(foo_test PROPERTIES CXX_STANDARD 11 CXX_STANDARD_REQUIRED ON CXX_EXTENSIONS OFF)
```

The code doesn't compile without setting the C++ standard in the `CMakeLists.txt` file

**Comment by ruslo** [comment]
> The code doesn't compile without setting the C++ standard in the CMakeLists.txt file

Setting C++ standard locally is not a good idea in general:
* http://cgold.readthedocs.io/en/latest/tutorials/libraries/symbols/odr-global.html

In Hunter testing C++ standard set in toolchain, no need to set it locally.

> Tested using

@maddouri See for local testing:
* https://docs.hunter.sh/en/latest/creating-new/create/cmake.html#testing-locally

**Comment by dvirtz** [comment]
also opened a https://github.com/ingenue/hunter/pull/161 to exclude failing toolchains.

File: examples/Catch/CMakeLists.txt [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
+
+set(SOURCES main.cpp
+            foo_test.cpp
+            foo.cpp)
+set(HEADERS foo.hpp)
+
+add_executable(foo_test ${SOURCES} ${HEADERS})
+target_link_libraries(foo_test PUBLIC Catch::Catch)
+
+enable_testing(true)
```
Comment (ruslo) [inline examples/Catch/CMakeLists.txt]: It's just `enable_testing()` not `enable_testing(true)`. Also we are not running tests so this one not needed here.

**Comment by dvirtz** [comment]
rebased and squashed

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2782
* https://travis-ci.org/ingenue/hunter/builds/300872055?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2783
* https://travis-ci.org/ingenue/hunter/builds/300935867?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.19.151

**Pull request merged**